### PR TITLE
Make the swupd update always be the layer two

### PR DESCRIPTION
**Pull request by qzheng527**
To help analyze the container image size by layers, assume
each container image be comprised of:
Layer 1: clearlinux:latest
Layer 2: swupd update
Other layers: the container configurations

With this, we can continue to optimize layer size accordingly.

Signed-off-by: Qi Zheng <qi.zheng@intel.com>

### CODE REVIEW

File: elasticsearch/Dockerfile [hunk 1]
```diff
@@ -3,8 +3,8 @@ MAINTAINER qi.zheng@intel.com
 
 ARG swupd_args
 
-RUN swupd update $swupd_args \
-	&& swupd bundle-add elasticsearch $swupd_args \
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add elasticsearch $swupd_args \
 	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline elasticsearch/Dockerfile]: We should remove the whole /var/lib/swupd directory as what's being done for the cgit Dockerfile

File: httpd/Dockerfile [hunk 2]
```diff
@@ -3,8 +3,8 @@ MAINTAINER qi.zheng@intel.com
 
 ARG swupd_args
 
-RUN swupd update $swupd_args \
-	&& swupd bundle-add httpd $swupd_args \
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add httpd $swupd_args \
 	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline httpd/Dockerfile]: Ditto for removing the whole directory rather than just the contents

File: iperf/Dockerfile [hunk 3]
```diff
@@ -3,8 +3,9 @@ MAINTAINER long1.wang@intel.com
 
 ARG swupd_args
 
-RUN swupd update --no-boot-update $swupd_args && \
-    swupd bundle-add iperf $swupd_args
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add iperf $swupd_args \
+	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline iperf/Dockerfile]: Ditto for removing the whole directory rather than just the contents

File: mariadb/Dockerfile [hunk 4]
```diff
@@ -3,8 +3,8 @@ MAINTAINER qi.zheng@intel.com
 
 ARG swupd_args
 
-RUN swupd update $swupd_args \
-	&& swupd bundle-add mariadb $swupd_args \
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add mariadb $swupd_args \
 	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline mariadb/Dockerfile]: Ditto for removing the whole directory rather than just the contents

File: memcached/Dockerfile [hunk 5]
```diff
@@ -3,8 +3,8 @@ MAINTAINER qi.zheng@intel.com
 
 ARG swupd_args
 
-RUN swupd update $swupd_args \
-	&& swupd bundle-add memcached $swupd_args \
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add memcached $swupd_args \
 	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline memcached/Dockerfile]: Ditto for removing the whole directory rather than just the contents

File: node/Dockerfile [hunk 6]
```diff
@@ -3,10 +3,8 @@ MAINTAINER jianjun.liu@intel.com
 
 ARG swupd_args
 
-RUN swupd update $swupd_args \
-	&& swupd bundle-add nodejs-basic $swupd_args \
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add nodejs-basic $swupd_args \
 	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline node/Dockerfile]: Ditto for removing the whole directory rather than just the contents

File: php/Dockerfile [hunk 7]
```diff
@@ -3,8 +3,8 @@ MAINTAINER jianjun.liu@intel.com
 
 ARG swupd_args
 
-RUN swupd update $swupd_args \
-	&& swupd bundle-add php-basic $swupd_args \
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add php-basic $swupd_args \
 	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline php/Dockerfile]: Ditto for removing the whole directory rather than just the contents

File: postgres/Dockerfile [hunk 8]
```diff
@@ -1,8 +1,11 @@
 FROM clearlinux:latest
 MAINTAINER william.douglas@intel.com
 
-RUN swupd update $swupd_args && \
-    swupd bundle-add postgresql $swupd_args
+ARG swupd_args
+
+RUN swupd update --no-boot-update $swupd_args
+RUN swupd bundle-add postgresql $swupd_args \
+	&& rm -rf /var/lib/swupd/*
```
Comment (gtkramer) [inline postgres/Dockerfile]: Ditto for removing the whole directory rather than just the contents

**Comment by gtkramer** [comment]
For all instances where we're doing a `rm -rf /var/lib/swupd` or `rm -rf /var/lib/swupd/*`, we should replace them all with `swupd clean --all` to remove an implementation detail that swupd manages from the Dockerfile.

**Comment by gtkramer** [comment]
Aside from needing to use `swupd clean`, this looks good.

**Comment by gtkramer** [comment]
@unrahul fyi about the `swupd clean` usage.

**Comment by qzheng527** [comment]
I tired the "swupd clean --all" and "swupd clean --dry-run" on my Clear dev machine, it cleaned nothing even if there were many contents on /var/lib/swupd/. I also did the test on running container clearlinux, did "swupd update" first to make sure there was something in /var/lib/swupd/, then did "swupd clean --all", still not cleaned everything.

**Comment by qzheng527** [comment]
@gtkramer 
For `rm -rf /var/lib/swupd` or `rm -rf /var/lib/swupd/*`, the clearlinux:latest uses the latter one. There is the folder /var/lib/swupd/ with empty content on container clearlinux:latest. I guess we should follow it to use `rm -rf /var/lib/swupd/*` for all the Dockerfiles

**Comment by liujianjunhq** [comment]
+1 lgtm

**Pull request merged**